pivotx/lib.php

```diff
@@ -4377,7 +4428,7 @@ function saveSerialize($filename, &$data) {
     // open the file and lock it.
     if($fp=fopen($filename, "a")) {
         
-        if (flock( $fp, LOCK_EX | LOCK_NB )) {
```

Comment:
Hmm maybe not such a good idea after all to make this lock a blocking call...

Reply:
Why? You think it's better to fail immediately, than to wait for the lock? Or for some othet reason?

I tend to agree in particular since saveSerialize is called at unexpected times, not only when an editor is working.

Reply:
Yea I was suddenly doubting my past self. 
I was thinking maybe there is a case I'm not aware of that could indefinitely block on that lock.
For instance, the original code seems to imply that flock() could fail if your file permission is incorrect.
But if that's the only instance where flock() would indefinitely be blocked surely php would time out. 
In this case it's more like the responsibility of the site's admin to fix that before putting the blog online. 

I'm mostly thinking out-loud but since it is critical even for viewing pages that `saveSerialize()` do not fail you'd want to wait for it until it works. The only alternative would be to implement some sort of automatic retrial with timeout, so why not just rely on php internal timeout right? 

If the site is so busy that the blocking call becomes an issue, it can be seen as either a limitation of the flat file db, or an a deeper architecture issue (we lack a mechanism to safely read files for **viewing** without flock()).

So I guess, after writing all this down, I'm in favor to my past self to make that a blocking call XD